src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+        const onChainLatestJoinChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const latestExitChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const exitChannelBlocks = this.agreementManager.getExitChannelChain();
 
-        const forkCntNumber = Number(forkCnt);
-        //TODO! stop signing for the current fork
-        this.setForkDisputed(forkCntNumber);
-        proofs.forEach((proof) => this.addProof(forkCntNumber, proof));
-        const _dispute = this.disputes.get(forkCntNumber);
-        if (!_dispute) {
-            await this.createNewDispute(
+        const {
+            encodedModifiedState,
+            exitBlock,
+            totalDeposits,
+            totalWithdrawals
+        } = (await this.stateChannelManagerContract.generateDisputeOutputState(
+            this.agreementManager.getStateMachineState(
                 forkCnt,
-                foldedParticipant,
-                foldedTransactionCnt,
-                proofs
-            );
-        }
+                transactionCnt
+            )!,
+            proofs,
+            { channelId: this.channelId },
+            onChainSlashes,
+            ethers.ZeroAddress,
+            ethers.ZeroAddress,
+            this.agreementManager.getJoinChannelChain(),
+            this.agreementManager.getSnapShot(forkCnt, transactionCnt)!
+        )) as unknown as DisputeOutputState;
+
+        const participants =
+            (await this.stateChannelManagerContract.getStatemachineParticipants(
+                encodedModifiedState
+            )) as unknown as AddressLike[];
 
-        const newDispute = await this.stateChannelManagerContract.getDispute(
-            this.channelId
+        const disputeOutputStateSnapshot: StateSnapshotStruct = {
+            stateMachineStateHash: ethers.keccak256(
+                ethers.toUtf8Bytes(encodedModifiedState)
+            ),
+            participants,
+            forkCnt: forkCnt + 1,
+            latestExitChannelBlockHash,
+            latestJoinChannelBlockHash: onChainLatestJoinChannelBlockHash,
+            totalDeposits,
+            totalWithdrawals
+        };
+        const disputeOutputStateSnapshotHash = ethers.keccak256(
+            EvmUtils.encodeStateSnapshot(disputeOutputStateSnapshot)
         );
-        //TODO! check newDispute 0000 bytes
-        if (newDispute.channelId == ethers.ZeroHash) {
-            throw new Error(
-                "DisputeHandler - createDispute - no dispute created"
-            );
-        }
-        await this.rechallengeRecursive(newDispute);
+
+        const { dispute: defaultDispute, timeout: defaultTimeout } =
+            this.createDefaultDisputeStruct();
+
+        const disputeAuditingData = this.createDisputeAuditingData(
+            forkCnt,
+            transactionCnt,
+            disputeOutputStateSnapshot,
+            timestamp,
+            defaultDispute,
```

Comment:
Don't understand the logic here - how does passing a defaultDispute for previousDispute make any sense